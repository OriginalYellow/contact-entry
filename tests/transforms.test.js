const allContacts = require('./all-contacts.mock.json')

const { 
  transformContact,
  transformContacts,
} = require('../transforms.js')

it('should return a correctly transformed contact', () => {
  const result = transformContact(allContacts[0])
  expect(result).toMatchSnapshot()
})

it('should return an array of correctly transformed contacts', () => {
  const result = transformContacts(allContacts)
  expect(result).toMatchSnapshot()
})
