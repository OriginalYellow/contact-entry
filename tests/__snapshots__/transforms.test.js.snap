// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a correctly transformed contact 1`] = `
Object {
  "address": Object {
    "city": "Cannon",
    "state": "Delaware",
    "street": "8360 High Autumn Row",
    "zip": "19797",
  },
  "email": "harold.gilkey@yahoo.com",
  "id": 1,
  "name": Object {
    "first": "Harold",
    "last": "Gilkey",
    "middle": "Francis",
  },
  "phone": Array [
    Object {
      "number": "302-611-9148",
      "type": "home",
    },
    Object {
      "number": "302-532-9427",
      "type": "mobile",
    },
  ],
}
`;

exports[`should return an array of correctly transformed contacts 1`] = `
Array [
  Object {
    "address": Object {
      "city": "Cannon",
      "state": "Delaware",
      "street": "8360 High Autumn Row",
      "zip": "19797",
    },
    "email": "harold.gilkey@yahoo.com",
    "id": 1,
    "name": Object {
      "first": "Harold",
      "last": "Gilkey",
      "middle": "Francis",
    },
    "phone": Array [
      Object {
        "number": "302-611-9148",
        "type": "home",
      },
      Object {
        "number": "302-532-9427",
        "type": "mobile",
      },
    ],
  },
  Object {
    "address": Object {
      "city": "West Brunswick",
      "state": "New Jersey",
      "street": "7 Coleman Rd",
      "zip": "08816",
    },
    "email": "michael.robert.joseph2@gmail.com",
    "id": 2,
    "name": Object {
      "first": "Michael",
      "last": "Joseph",
      "middle": "Robert",
    },
    "phone": Array [
      Object {
        "number": "732-614-6946",
        "type": "home",
      },
      Object {
        "number": "302-432-8272",
        "type": "mobile",
      },
    ],
  },
]
`;
